contracts/rust/deployer/src/lib.rs

```diff
@@ -1046,6 +1206,86 @@ mod tests {
         test_upgrade_light_client_to_v2_helper(true).await
     }
 
+    // This test is used to test the upgrade of the LightClientProxy via the multisig wallet
+    // It only tests the upgrade proposal via the typescript script and thus requires the upgrade proposal to be sent to a real network
+    // However, the contracts are deployed on anvil, so the test will pass even if the upgrade proposal is not executed
+    // The test assumes that there is a file .env.deployer.rs.test in the root directory with the following variables:
+    // RPC_URL=
+    // SAFE_MULTISIG_ADDRESS=0x0000000000000000000000000000000000000000
+    // SAFE_ORCHESTRATOR_PRIVATE_KEY=0x0000000000000000000000000000000000000000000000000000000000000000
+    // Ensure that the private key has proposal rights on the Safe Multisig Wallet and the SDK supports the network
+    async fn test_upgrade_light_client_to_v2_multisig_owner_helper(is_mock: bool) -> Result<()> {
+        dotenvy::from_filename_override(".env.deployer.rs.test").ok();
```

Comment:
We shouldn't manipulate the environment from tests. `cargo test` doesn't use a single process per test (nextest does though) so it can lead to weird things happening.

I think we can use https://docs.rs/dotenvy/latest/dotenvy/fn.from_filename_iter.html convert that into a HashMap and read the values from there instead of loading it into the environment and reading them back.

Reply:
thank you for that https://github.com/EspressoSystems/espresso-network/commit/f0a148fe64314e44205e5a21be00aa9fbbcfa06a